### ISSUE TITLE
Ghost button in latest dev build

### ISSUE THREAD

**Issue by jmacura** [Quick-Event/quickbox#774]
There is a blank button in the top control panel which opens another blank drop-down.
![image](https://user-images.githubusercontent.com/5598693/178161489-c26c789a-2d66-4eca-81eb-8bf215f271f7.png)
![image](https://user-images.githubusercontent.com/5598693/178161494-c5218ca8-b4aa-4eb7-a638-40408a9376ed.png)
Nothing special in the log.
![image](https://user-images.githubusercontent.com/5598693/178161518-bd70f245-c20b-4994-a046-a6c496085dda.png)

Fresh new install. No event exists.

---
QE version: https://github.com/Quick-Event/quickbox/actions/runs/2556216254
OS version: Win 10 Pro 21H1

**Comment by paukert** [comment]
Podobných chyb je v QE bohužel více - není ošetřen stav, kdy není otevřen žádný závod. Třeba po kliknutí na tlačítko přidání závodníka QE spadne, další chyby očekávám i při pokusu o export atd.

**Comment by jmacura** [comment]
Chápu. Na druhou stranu... ošetření chyb typu "nezobrazuj toto, když neexistuje tamto" zabere člověku, který se v kódu vyzná, asi tak... 5 minut? Včetně spuštění IDE a kompilace apky? To je docela "low-hanging fruit", ne? 😉 

A pro BFU to má přínos nepopsatelný.

**Comment by otahirs** [comment]
Těch míst kde je to třeba ošetřit je hodně. Uživateli vyskočí upozornění že má nejdřív otevřít/vytvořit/importovat event. V menu jsou zakázané ostatní volby, aby to měl jednoduší.
Je to edge case, a jiných issues je víc než dost.

**Comment by fvacek** [comment]
Jsem rad, ze uz se dostavame k reseni takovych corner-cases, znamena to, ze zbytek je funkcni, protoze uzivatele repotuji vzdy nejvetsi problem. Chyba to samozrejme je, jenom nesouhlasim s `low-hanging frui`. Tech veci, ktere postradaji smysl, kdyz neni otevren zadny zavod je mnohem vic a urcite je nema smysl resit pomoci nejakych 5-ti minutovych quick fixu. Pokud neni otevren zadny zavod, jsou ghost buttony ve vsech perspektivach a to nemluvim o prazdnych tabulkach, atd.

Takze pokud chceme resit toto, asi by bylo nejlepsi, nedat uzivateli moznost opustit connection wizard , dokud neotevre nebo nenaimportuje nejaky zavot a to urcite neni na 5 minut.

V kazdem pripade, jsem rad, ze se toto tema otevira. Zpusob pripojeni k databazi a otevreni zavodu vznikl dost zivelne a nabranil bych se nejakemu smysluplnejsimu redesignu, ktery by treba vyresil i ghost buttony.

Spravne reseni je asi

**Comment by jmacura** [comment]
> Jsem rad, ze uz se dostavame k reseni takovych corner-cases, znamena to, ze zbytek je funkcni, protoze uzivatele repotuji vzdy nejvetsi problem. 

Nebo to znamená, že uživatelé se k větším problémům ani nedostanou.
Z hlediska UX:
1. BFU spustí program
2. "tady je něco, co nic nedělá"
3. "je to rozbitý"
4. "tak tohle nechci"
5. game over

**Comment by fvacek** [comment]
Ano, tohle je mozny scenar, pokousime se prozatim cilit na chytrejsi uzivatele, na usporadani zavodu je to beztak potreba. Jinak se ale mozne uprave/vylepseni vubec nevyhybam, viz muj predchozi post.
